### PR TITLE
Tsify llvm-ast

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
Baseline for converting lib/llvm-ast, I have not tested locally yet

### CODE REVIEW

File: lib/llvm-ast.ts [hunk 1]
```diff
@@ -22,24 +22,37 @@
...
+import {ResultLine} from '../types/resultline/resultline.interfaces';
+
+// Almost every line of AST includes a span of related source lines:
+// In different forms like <line:a:b, line:c:d>
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace LocTypes {
+    export const NONE = 'none'; // No location specified
+    export const POINT = 'point'; // A single location: beginning of a token
+    export const SPAN = 'span'; // Two locations: first token to last token (beginning)
+}
```
Comment (jeremy-rifkin) [inline lib/llvm-ast.ts:37]: Moved to a namespace so the literal values are known for later

File: lib/llvm-ast.ts [hunk 2]
```diff
@@ -62,43 +81,44 @@ export class LlvmAstParser {
...
                     lfrom = span.begin;
                     lto = span.end;
                     break;
                 }
             }
-            if (span.type !== this.locTypes.NONE) {
-                line.source = {from: lfrom, to: lto};
+            if (span.type !== LocTypes.NONE) {
+                // TODO: ResultLineSource doesn't have to/from
+                (line.source as any) = {from: lfrom, to: lto};
```
Comment (jeremy-rifkin) [inline lib/llvm-ast.ts:115]: Not sure if this is wrong or if `ResultLineSource` is wrong

**Review by mattgodbolt — COMMENTED** [review]
needs a `npm run lint` (and a local test :D )

File: lib/llvm-ast.ts (hunk 2 above)
Comment (mattgodbolt) [reply lib/llvm-ast.ts:114]: oh hmm that's weird...another typing mystery.

**Comment by jeremy-rifkin** [comment]
Thank you, completely missed those warnings :) Tested locally and looks good to go

**Pull request merged**